app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,9 @@
+class Admin::BaseController < ApplicationController
+
+  before_action :require_admin
+
+  def require_admin
+    render file: "/public/404" unless current_user && current_user.admin?
```

Comment:
no specific message on the 404 page to let the user know what happened. 
the 404 message displays with the navbar above it, so i'm pretty sure a flash message will display if we create one, without having to change the redirect path. 
though, i'm not against redirecting to root if that's what guys think is better